### PR TITLE
Install manifests for rollback after update --from-rollback-file

### PR DESCRIPTION
**Pull request by Forgind**
Fixes https://github.com/dotnet/sdk/issues/37958 for 1xx

**Description**
Manifest updates were disabled for Install if the user had previously updated --from-rollback-file, as that clearly indicates they wanted to pin to a specific set of versions, but that did not take into consideration if a user wanted to install --from-rollback-file, which similarly indicates they want a specific version but a different specific version. This takes that into account.

**Customer Impact**
Customers who called `dotnet workload update --from-rollback-file <file>` then `dotnet workload install --from-rollback-file <file>` would have the second command not install the desired versions.

**Regression**?
Yes. The decision to update or not update is made based on the install state file that was only added in 8.0.1xx.

**Risk**
Low

**Tests**
TBD

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs [hunk 1]
```diff
@@ -134,7 +134,7 @@ public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManif
             if (!skipManifestUpdate)
             {
                 var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
-                if (File.Exists(installStateFilePath))
```
Comment (marcpopMSFT) [inline src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs:137]: I'm wondering why we don't move line 160 up to 132 and just use "!useRollback" here? Seems like the code may have been trying to save the isnull check but I can't imagine that's a slow check.

Comment (Forgind) [reply src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs:137]: Yeah; I can do that. I typically try to change as little as possible for servicing changes, but that shouldn't hurt anything.

Comment (joeloff) [reply src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs:137]: Just take note that on line 168, the default `useRollback` value is passed through and may not have been set based on if the `skipManifestUpdate` option that controls the conditional block on line 145.

Comment (Forgind) [reply src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs:137]: joeloff is right, and in particular, deciding to skip the manifest update isn't decided yet on line 132.

The useRollback parameter passed into InstallWorkloadsWithInstallRecord is only used to modify the install state. If we legitimately rolled back to a specific set of manifests, it is created with those manifests. If not, it is deleted (in 8.0.1xx). Moving useRollback up may make it create the install state when it didn't update manifests at all, much less from a rollback file.

**Review by marcpopMSFT — APPROVED** [review]

**Pull request merged**